File: backend/src/sidra/api/models/maintenance_status.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class MaintenanceStatus(BaseModel):
    """What the Maintenance screen shows before he presses anything."""

    catalog_seeded: bool
    ledger_seeded: bool
    works: int
    stored_units: int
    tracks: int
    advances: int

    ledger_exported_at: datetime | None
    """When ``data/ledger.json`` was last written. None when it has never been."""

    safety_copy_at: datetime | None
    """When a correction last wrote its safety copy. None when none ever has."""

File: frontend/src/types/MaintenanceStatus.ts
/** What the Maintenance screen shows before he presses anything. */
export interface MaintenanceStatus {
  readonly catalog_seeded: boolean;
  readonly ledger_seeded: boolean;
  readonly works: number;
  readonly stored_units: number;
  readonly tracks: number;
  readonly advances: number;
  /** When data/ledger.json was last written. Null when it never has been. */
  readonly ledger_exported_at: string | null;
  /** When a correction last wrote its safety copy. Null when none ever has. */
  readonly safety_copy_at: string | null;
}
